plugins/notifications-backend-module-slack/src/lib/SlackNotificationProcessor.ts

```diff
@@ -0,0 +1,268 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AuthService,
+  CacheService,
+  DiscoveryService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { CatalogClient } from '@backstage/catalog-client';
+import {
+  Entity,
+  GroupEntity,
+  UserEntity,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { Notification } from '@backstage/plugin-notifications-common';
+import {
+  NotificationProcessor,
+  NotificationSendOptions,
+} from '@backstage/plugin-notifications-node';
+
+export const ANNOTATION_SLACK_CHANNEL_ID = 'slack.com/channel-id';
+const DEFAULT_CACHE_TTL = 1000 * 60 * 60; // 1 hour
+
+export class SlackNotificationProcessor implements NotificationProcessor {
+  private readonly logger: LoggerService;
+  private readonly directWebhookUrl?: string;
+  private readonly channelWebhookUrl?: string;
+  private readonly cache: CacheService;
+  private readonly ttl: number;
+  private readonly catalogClient: CatalogClient;
+  private readonly auth: AuthService;
+
+  static fromConfig(
+    config: Config,
+    options: {
+      auth: AuthService;
+      discovery: DiscoveryService;
+      logger: LoggerService;
+      cache: CacheService;
+      catalogClient?: CatalogClient; // used for testing
+    },
+  ): SlackNotificationProcessor[] {
+    const slackConfig =
+      config.getOptionalConfigArray('notifications.processors.slack') ?? [];
+    return slackConfig.map(c => {
+      const directWebhookUrl = c.getOptionalString('directWebhookUrl');
+      const channelWebhookUrl = c.getOptionalString('channelWebhookUrl');
+      const ttl = c.getOptionalNumber('cache.ttl') ?? DEFAULT_CACHE_TTL;
+      return new SlackNotificationProcessor({
+        directWebhookUrl,
+        channelWebhookUrl,
+        ttl,
+        ...options,
+      });
+    });
+  }
+
+  private constructor(options: {
+    auth: AuthService;
+    discovery: DiscoveryService;
+    logger: LoggerService;
+    directWebhookUrl?: string;
+    channelWebhookUrl?: string;
+    cache: CacheService;
+    ttl: number;
+    catalogClient?: CatalogClient;
+  }) {
+    const { auth, directWebhookUrl, channelWebhookUrl, logger, cache, ttl } =
+      options;
+    this.logger = logger;
+    this.directWebhookUrl = directWebhookUrl;
+    this.channelWebhookUrl = channelWebhookUrl;
+    this.cache = cache;
+    this.ttl = ttl;
+    this.auth = auth;
+    this.catalogClient =
+      options.catalogClient ??
+      new CatalogClient({
+        discoveryApi: options.discovery,
+      });
+  }
+  getName(): string {
+    return 'Slack';
+  }
+
+  async processOptions(
+    options: NotificationSendOptions,
+  ): Promise<NotificationSendOptions> {
+    if (!this.channelWebhookUrl || options.recipients.type === 'broadcast') {
+      // Does not support broadcast notifications
```

Comment:
Maybe something to consider for future how to support these (some specific channel from config, or fetching all users or something?)

Reply:
Maybe a dedicated broadcast channel? TBH I don't think I have a use case for broadcast slack notifications as they tend to become a nuisance if often. Maybe for the announcements plugin could be useful.

Reply:
I think it'd be fine to add broadcast support in a follow-on, esp if its requested.

Reply:
Yeah, let's leave it for now like this and add support later if requested. Dedicated channel could be an option, maybe through configuration how the broadcast messages should be handled (to all users / dedicated channel / ignore)

Reply:
I made broadcast config support for email here: https://github.com/backstage/backstage/pull/24322 

Maybe something to consider for this as well. Later is ok too.

Reply:
I think support for broadcast can be added later.
just a nitpick (feel free to ignore) for this line of code:
```suggestion
       if (!this.channelWebhookUrl || options.recipients.type !== 'entity') {
```

No difference now but it's safe for later.